traefik/oam.go

```diff
@@ -0,0 +1,185 @@
+package traefik
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/layer5io/meshkit/models/oam/core/v1alpha1"
+	"gopkg.in/yaml.v2"
+)
+
+// CompHandler is the type for functions which can handle OAM components
+type CompHandler func(*Mesh, v1alpha1.Component, bool) (string, error)
+
+// HandleComponents handles the processing of OAM components
+func (mesh *Mesh) HandleComponents(comps []v1alpha1.Component, isDel bool) (string, error) {
+	var errs []error
+	var msgs []string
+
+	compFuncMap := map[string]CompHandler{
+		"TraefikMesh": handleComponentTraefikMesh,
+	}
+
+	for _, comp := range comps {
+		fnc, ok := compFuncMap[comp.Spec.Type]
+		if !ok {
+			msg, err := handleTraefikCoreComponent(mesh, comp, isDel, "", "")
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+
+			msgs = append(msgs, msg)
+			continue
+		}
+
+		msg, err := fnc(mesh, comp, isDel)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		msgs = append(msgs, msg)
+	}
+
+	if err := mergeErrors(errs); err != nil {
+		return mergeMsgs(msgs), err
+	}
+
+	return mergeMsgs(msgs), nil
+}
+
+// HandleApplicationConfiguration handles the processing of OAM application configuration
+func (mesh *Mesh) HandleApplicationConfiguration(config v1alpha1.Configuration, isDel bool) (string, error) {
+	var errs []error
+	var msgs []string
+	for _, comp := range config.Spec.Components {
+		for _, trait := range comp.Traits {
+			if trait.Name == "automaticSidecarInjection.Traefik" {
```

Comment:
This trait should not exist. There is not sidecar injection in this mesh.

Reply:
Removed